src/core/find.js

```diff
@@ -186,7 +187,16 @@ module.exports = (haystack_, searches_, ctx) => {
             }
           }
         }
-        stack.push(false, searchesOut, isArray ? Number(key) : key, depth + 1);
+        if (ctx.orderByNeedles) {
+          searchesOut.index = Buffer.from(searchesOut.map((e) => getOrder(e)).sort());
```

Comment:
```suggestion
          searchesOut.index = Buffer.from(searchesOut.map((e) => getOrder(e)).sort());
```
why do we need to sort this?

Reply:
Because different searches are pursued in parallel